src/components/legend/index.js

```diff
@@ -663,14 +686,53 @@ legend.repositionLegend = function(td, traces){
     lx = Math.round(lx);
     ly = Math.round(ly);
 
-    fullLayout._infolayer.selectAll('svg.legend')
-        .call(Plotly.Drawing.setRect, lx, ly, legendwidth, legendheight);
-    fullLayout._infolayer.selectAll('svg.legend .bg')
-        .call(Plotly.Drawing.setRect, borderwidth/2, borderwidth/2,
-            legendwidth-borderwidth, legendheight-borderwidth);
+
+    var legendsvg = fullLayout._infolayer.selectAll('svg.legend'),
```

Comment:
This section should likely be moved to the `draw` step, but everything is highly coupled right now and the order of drawing/sizing/positioning is very scattered.